client/state/plugins/wporg/selectors.js

```diff
@@ -24,4 +24,36 @@ const isFetched = function( state, pluginSlug ) {
 	return !! plugin.fetched;
 };
 
-export default { getPlugin, isFetching, isFetched };
+const canFetchList = function( state, category, page, searchTerm ) {
+	const { lists } = state.plugins.wporg;
+	if ( searchTerm && lists.fetching.search ) {
+		return false;
+	}
+	if ( lists.fetching[ category ] ) {
+		return false;
+	}
+
+	if ( lists.lastFetchedPage[ category ] >= page ) {
+		return false;
+	}
+
+	return true;
```

Comment:
another way to look at this is a simple list where any truth value means the whole thing fails…

```js
const canFetchList = ( state, category, page, searchTerm ) => {
	const { lists } = state.plugins.wporg;

	// if any of these rules are true
	// then we can't fetch
	return ! [
		searchTerm && lists.fetching.search ),
		lists.fetching[ category ],
		lists.lastFetchedPage[ category ] >= page,
	].some( identity );
}
```